src/test/java/tests/DetailMessage.java

```diff
@@ -71,30 +55,30 @@ final class DetailMessage extends TestDiagnostic {
   final String readableMessage;
 
   /**
-   * Returns an object parsed from a diagnostic message, or {@code null} if the message doesn't
-   * match the expected format.
+   * Returns an object parsed from a diagnostic message.
    *
    * @param rootDirectory if not null, a root directory prefix to remove from the file part of the
    *     message
    */
-  static @Nullable DetailMessage parse(String input, @Nullable Path rootDirectory) {
-    Matcher messageMatcher = MESSAGE_PATTERN.matcher(input);
-    if (!messageMatcher.matches()) {
-      return null;
-    }
-
-    Path file = Paths.get(messageMatcher.group("file"));
-    if (rootDirectory != null) {
+  static DetailMessage parse(TestDiagnostic input, @Nullable Path rootDirectory) {
+    Path file = input.getFile();
+    if (rootDirectory != null && !file.toString().equals("")) {
```

Comment:
When would `file` be empty?

Reply:
I can confirm that it was happening, at least :) I didn't investigate further, but I assumed it came from this:

https://github.com/jspecify/jspecify-reference-checker/blob/fae43cfec3b0d9e70cb02ef2d73fa7e05653d4b6/src/main/java/com/google/jspecify/nullness/NullSpecChecker.java#L83-L90

Reply:
Every test that has an error also produces a note:

````
Note: 
  =====
  The nullness checker is still early in development.
  It has many rough edges.
  For an introduction, see https://jspecify.dev/docs/user-guide
  =====
````

That diagnostics has no file path.